gqt_test.go

```diff
@@ -7,6 +7,363 @@ import (
+			value: gqt.ValueObject{
+				Fields: []gqt.ObjectField{
+					{
+						Name: "a",
+						Value: gqt.ConstraintValLessOrEqual{
+							Value: 42.0,
+						},
+					},
+				},
+			},
+		},
+		{
+			input: gqt.ConstraintValNotEqual{
+				Value: gqt.ValueArray{
+					Items: []gqt.Constraint{
+						gqt.ConstraintTypeEqual{
+							TypeName: "Float",
+						},
+					},
+				},
+			},
+			name: "ConstraintValNotEqual",
+			value: gqt.ValueArray{
+				Items: []gqt.Constraint{
+					gqt.ConstraintTypeEqual{
+						TypeName: "Float",
+					},
+				},
+			},
+		},
+		{
+			input: gqt.ConstraintValGreater{
+				Value: 42.0,
+			},
+			name:  "ConstraintValGreater",
+			value: 42.0,
+		},
+		{
+			input: gqt.ConstraintValLess{
+				Value: 42.0,
+			},
+			name:  "ConstraintValLess",
+			value: 42.0,
+		},
+		{
+			input: gqt.ConstraintValGreaterOrEqual{
+				Value: 69.0,
+			},
+			name:  "ConstraintValGreaterOrEqual",
+			value: 69.0,
+		},
+		{
+			input: gqt.ConstraintValLessOrEqual{
+				Value: 69.0,
+			},
+			name:  "ConstraintValLessOrEqual",
+			value: 69.0,
+		},
+		{
+			input: gqt.ConstraintBytelenEqual{
+				Value: 1984,
+			},
+			name:  "ConstraintBytelenEqual",
+			value: uint(1984),
+		},
+		{
+			input: gqt.ConstraintBytelenNotEqual{
+				Value: 1984,
+			},
+			name:  "ConstraintBytelenNotEqual",
+			value: uint(1984),
+		},
+		{
+			input: gqt.ConstraintBytelenGreater{
+				Value: 282,
+			},
+			name:  "ConstraintBytelenGreater",
+			value: uint(282),
+		},
+		{
+			input: gqt.ConstraintBytelenLess{
+				Value: 282,
+			},
+			name:  "ConstraintBytelenLess",
+			value: uint(282),
+		},
+		{
+			input: gqt.ConstraintBytelenGreaterOrEqual{
+				Value: 27015,
+			},
+			name:  "ConstraintBytelenGreaterOrEqual",
+			value: uint(27015),
+		},
+		{
+			input: gqt.ConstraintBytelenLessOrEqual{
+				Value: 27015,
+			},
+			name:  "ConstraintBytelenLessOrEqual",
+			value: uint(27015),
+		},
+		{
+			input: gqt.ConstraintLenEqual{
+				Value: 997,
+			},
+			name:  "ConstraintLenEqual",
+			value: uint(997),
+		},
+		{
+			input: gqt.ConstraintLenNotEqual{
+				Value: 997,
+			},
+			name:  "ConstraintLenNotEqual",
+			value: uint(997),
+		},
+		{
+			input: gqt.ConstraintLenGreater{
+				Value: 47,
+			},
+			name:  "ConstraintLenGreater",
+			value: uint(47),
+		},
+		{
+			input: gqt.ConstraintLenLess{
+				Value: 47,
+			},
+			name:  "ConstraintLenLess",
+			value: uint(47),
+		},
+		{
+			input: gqt.ConstraintLenGreaterOrEqual{
+				Value: 404,
+			},
+			name:  "ConstraintLenGreaterOrEqual",
+			value: uint(404),
+		},
+		{
+			input: gqt.ConstraintLenLessOrEqual{
+				Value: 404,
+			},
+			name:  "ConstraintLenLessOrEqual",
+			value: uint(404),
+		},
+	} {
+		t.Run(td.name, func(t *testing.T) {
+			name, value, err := gqt.ConstraintNameAndValue(td.input)
+			require.Nil(t, err)
+			require.Equal(t, td.name, name)
+			require.Equal(t, td.value, value)
+		})
+	}
+}
+
+func TestConstraintName(t *testing.T) {
```

Comment:
this test is missing a case where `input: nil; expect: ""`

Reply:
Why do we need this test case?

Reply:
because you want to cover this case as well 😄

Reply:
I mean it won't even work, as I use generics in this unit test, so `nil` doesn't have `Name()` function for sure)

Reply:
Ah, you're calling an interface method. Right, then it doesn't make sense.